sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2505,12 +2505,14 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
         outer: LogicalPlan)(
         f: (LogicalPlan, Seq[Expression]) => SubqueryExpression): SubqueryExpression = {
       val newSubqueryPlan = AnalysisContext.withOuterPlan(outer) {
-        executeSameContext(e.plan)
+        val analyzed = executeSameContext(e.plan)
+        checkAnalysis(analyzed)
```

Comment:
I see the problem now. Because of CTE, we want to check the analysis for the main query and subqueries at the same time. But now we have to check it earlier for subqueries.

Reply:
My suggestion:
1. do not check analysis here, but only set the analyzed flag.
2. in `CheckAnalysis`, set the subquery plan as not `analyzed` before checking it.

Reply:
Thanks for your suggestions, I will try the first, for the second I still need to spend some time understanding it.